vars/neoDeploy.groovy

```diff
@@ -3,49 +3,114 @@ import com.sap.piper.Utils
+            }
+            warAction = utils.getMandatoryParameter(parameters, 'warAction', 'deploy')
+        }
+
+        def applicationName
+        def runtime
+        def runtimeVersion
+        def vmSize
+        if (deployMode == 'WAR_PARAMS') {
+            applicationName = utils.getMandatoryParameter(parameters, 'applicationName', null)
+            runtime = utils.getMandatoryParameter(parameters, 'runtime', null)
+            runtimeVersion = utils.getMandatoryParameter(parameters, 'runtimeVersion', null)
+            vmSize = utils.getMandatoryParameter(parameters, 'vmSize', 'lite')
+            warAction = utils.getMandatoryParameter(parameters, 'warAction', 'deploy')
+        }
+
         def defaultDeployHost = script.commonPipelineEnvironment.getConfigProperty('DEPLOY_HOST')
         def defaultDeployAccount = script.commonPipelineEnvironment.getConfigProperty('CI_DEPLOY_ACCOUNT')
         def defaultCredentialsId = script.commonPipelineEnvironment.getConfigProperty('neoCredentialsId')
         if (defaultCredentialsId == null) {
             defaultCredentialsId = 'CI_CREDENTIALS_ID'
         }
 
-        def deployHost = utils.getMandatoryParameter(parameters, 'deployHost', defaultDeployHost)
-        def deployAccount = utils.getMandatoryParameter(parameters, 'deployAccount', defaultDeployAccount)
+        def deployHost
+        def deployAccount
+        if (deployMode.equals('MTA') || deployMode.equals('WAR_PARAMS')) {
+            deployHost = utils.getMandatoryParameter(parameters, 'deployHost', defaultDeployHost)
+            deployAccount = utils.getMandatoryParameter(parameters, 'deployAccount', defaultDeployAccount)
+        }
+
         def credentialsId = parameters.get('neoCredentialsId', defaultCredentialsId)
 
         def neoExecutable = getNeoExecutable(parameters)
 
-        withCredentials([usernamePassword(
+        if (deployMode == 'MTA') {
+            withCredentials([usernamePassword(
                 credentialsId: credentialsId,
                 passwordVariable: 'password',
-                usernameVariable: 'username'
-        )]) {
-            sh """#!/bin/bash
-                    "${neoExecutable}" deploy-mta \
+                usernameVariable: 'username')]) {
+                sh """#!/bin/bash
+                      "${neoExecutable}" deploy-mta \
                       --user '${username}' \
                       --host '${deployHost}' \
                       --source "${archivePath.getAbsolutePath()}" \
                       --account '${deployAccount}' \
                       --password '${password}' \
                       --synchronous
-               """
+                   """
+            }
+        }
+
+        if (deployMode == 'WAR_PARAMS') {
+            withCredentials([usernamePassword(
+                credentialsId: credentialsId,
+                passwordVariable: 'password',
+                usernameVariable: 'username')]) {
+                sh """#!/bin/bash
```

Comment:
In order to avoid redundant repetitive coding it might be desirable to create first a string containing the common parts of the neo statement (e.g. shebang, neoExecutable, (command as variable), user, password, source) in all cases (host, account if not based on the properties file). IMO with this approach the code would be better maintainable.